go/store/nbs/journal_inspect.go

```diff
@@ -592,3 +657,106 @@ func (w *LineLogger) flush() {
 
 // Flush can be called manually to force any remaining bytes out.
 func (w *LineLogger) Flush() { w.flush() }
+
+// filterJournalFile creates a filtered copy of the journal that excludes records with the specified hashes by type.
+//
+// The |journalPath| is the path to the original journal file, used to create output file journalPath+".filtered".
+// The |buf| is the contents of the original journal file. The |filterRoots| and |filterChunks| are the lists of hashes
+// to exclude from the output journal file. Having 1 or more hashes to filter is expected.
+//
+// Returns exit code 0 on success, 1 on error.
+func filterJournalFile(journalPath string, buf []byte, filterRoots, filterChunks []hash.Hash) int {
+	dir := filepath.Dir(journalPath)
+	base := filepath.Base(journalPath)
+	outputPath := filepath.Join(dir, base+".filtered")
+
+	outputFile, err := os.OpenFile(outputPath, os.O_CREATE|os.O_WRONLY|os.O_EXCL, 0644)
+	if err != nil {
+		logrus.Errorf("Error creating filtered journal file: %v", err)
+		return 1
+	}
+	defer outputFile.Close()
+
+	_, filteredRecords, exitStatus := filterJournalCore(buf, outputFile, filterRoots, filterChunks)
+	if exitStatus != 0 {
+		return exitStatus
+	}
+
+	if filteredRecords == 0 {
+		logrus.Infof("No records matched the filter criteria. No changes made.")
+		return 1
+	}
+
+	// Print shell commands to replace the journal file
+	now := time.Now()
+	dateString := now.Format("2006_01_02_150405")
+
+	logrus.Infof("")
+	logrus.Infof("Filtered file: %s", outputPath)
+	logrus.Infof("")
+	logrus.Infof("To replace the original journal file, run these commands:")
+	logrus.Infof("cp %s %s_saved_%s", journalPath, journalPath, dateString)
+	logrus.Infof("mv %s %s", outputPath, journalPath)
+	logrus.Infof("rm %s", filepath.Join(filepath.Dir(journalPath), "journal.idx"))
```

Comment:
I know this is all adhoc testing code and stuff, but this kinda UI affordance stuff really belongs outside of `nbs`. Better to return enough structure from this function for the command implementation to handle presentation.